### PR TITLE
Fully --!strict and type annotated!

### PR DESCRIPTION
**Pull request by MaximumADHD**
It's backwards compatible with the current version! Just some small tweaks to make Luau happy, plus some touch ups for quality of life. Only breaking change is making `WindShake:Connect` a private function... but there was never any reason for it to be used outside of the module itself. I don't anticipate this will cause any issues.

### CODE REVIEW

File: src/init.lua [hunk 1]
```diff
@@ -20,55 +21,72 @@ local COLLECTION_TAG = "WindShake" -- The CollectionService tag to be watched an
...
-	assert(typeof(callback) == "function", "Unknown function: " .. funcName)
-
+local function Connect<Args...>(self: WindShake, event: RBXScriptSignal, callback: (self: WindShake, Args...) -> ()): RBXScriptConnection
 	return event:Connect(function(...)
 		return callback(self, ...)
 	end)
 end
 
-function WindShake:AddObjectShake(object: BasePart | Bone, settingsTable: WindShakeSettings?)
+function WindShake.AddObjectShake(self: WindShake, object: Instance, settingsTable: WindShakeSettings?)
```
Comment (boatbomber) [inline src/init.lua]: Why the change from `BasePart | Bone`?

File: src/init.lua [hunk 2]
```diff
@@ -239,12 +271,12 @@ function WindShake:Update(deltaTime: number)
 
 	debug.profileend()
 
-	workspace:BulkMoveTo(partList, cframeList, Enum.BulkMoveMode.FireCFrameChanged)
+	workspace:BulkMoveTo(partList :: { any }, cframeList, Enum.BulkMoveMode.FireCFrameChanged)
```
Comment (boatbomber) [inline src/init.lua]: Does this cast need to be here? Isn't `partList` already typed as `{ any }` by `WindShake._partList`?

File: src/init.lua (hunk 1 above)
Comment (MaximumADHD) [reply src/init.lua]: It's because the function was being directly plugged into the CollectionService callback, so it technically *was* consuming types of `Instance`. I can correct this behavior in the `Init` function.

File: src/init.lua (hunk 2 above)
Comment (MaximumADHD) [reply src/init.lua]: Yep. I did actually fix this in my fork already. Guess I forgot to commit it <_>
